sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/metric/MeterMetric.java

```diff
@@ -0,0 +1,144 @@
+        MeterRegistry.Config config = this.registry.config();
+        config.commonTags(
+                "agent", "sermant",
+                "agent.app.name", BootArgsIndexer.getAppName()
+        );
+        config.meterFilter(MeterFilter.maximumAllowableMetrics(metricConfig.getMaxTimeSeries()));
+    }
+
+    @Override
+    public Counter counter(String metricName, Tags tags, String description) {
+        io.micrometer.core.instrument.Counter counter = io.micrometer.core.instrument.Counter.builder(metricName)
+                .tags(getMeterTags(tags))
+                .description(description)
+                .register(registry);
+        return new MeterCounter(counter);
+    }
+
+    @Override
+    public Gauge gauge(String metricName, Tags tags, String description) {
+        return new MeterGauge(registry, metricName, getMeterTags(tags), description);
+    }
+
+    @Override
+    public Timer timer(String metricName, Tags tags, String description) {
+        io.micrometer.core.instrument.Timer timer = io.micrometer.core.instrument.Timer.builder(metricName)
+                .tags(getMeterTags(tags)).description(description).register(registry);
+        return new MeterTimer(registry, timer);
+    }
+
+    @Override
+    public Summary summary(String metricName, Tags tags, String description,
+            DistributionStatisticConfig distributionStatisticConfig) {
+        Builder summaryBuilder = DistributionSummary.builder(metricName)
+                .tags(getMeterTags(tags))
+                .description(description);
+
+        if (distributionStatisticConfig.getPercentileHistogram() != null) {
+            summaryBuilder.publishPercentileHistogram(distributionStatisticConfig.getPercentileHistogram());
+        }
+        if (distributionStatisticConfig.getPercentiles() != null) {
+            summaryBuilder.publishPercentiles(distributionStatisticConfig.getPercentiles());
+        }
+        if (distributionStatisticConfig.getPercentilePrecision() != null) {
+            summaryBuilder.percentilePrecision(distributionStatisticConfig.getPercentilePrecision());
+        }
+        if (distributionStatisticConfig.getMinimumExpectedValue() != null) {
+            summaryBuilder.minimumExpectedValue(distributionStatisticConfig.getMinimumExpectedValue());
+        }
+        if (distributionStatisticConfig.getMaximumExpectedValue() != null) {
+            summaryBuilder.maximumExpectedValue(distributionStatisticConfig.getMaximumExpectedValue());
+        }
+        if (distributionStatisticConfig.getExpiry() != null) {
+            summaryBuilder.distributionStatisticExpiry(distributionStatisticConfig.getExpiry());
+        }
+        if (distributionStatisticConfig.getBufferLength() != null) {
+            summaryBuilder.distributionStatisticBufferLength(distributionStatisticConfig.getBufferLength());
+        }
+        DistributionSummary summary = summaryBuilder.register(registry);
```

Comment:
Optional.ofNullable(distributionStatisticConfig.getPercentileHistogram()).ifPresent(summaryBuilder::publishPercentileHistogram);
 may be more clear

Reply:
fixed